src/app/shared/tabview-x/tabview-index-tracker.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Location } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TabviewIndexTrackerService {
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  private readonly FIRST_TAB = 0;
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  private readonly DEFAULT_QUERY_PARAM_KEY = 'tab';
```

Comment:
I think disabling these naming convention rules is a bad idea. Either we adjust the rules, or we comply with them.

Also, I think it would be more suiting to place these above the `@Injectable` statement as `const`'s instead. But that's just a personal preference of mine, as I dislike the `this.` prefix used in the param definitions below.

Reply:
Do we have any set standard for marking up class constants?

Reply:
No, I don't think so